File: server/src/test/scala/kpn/api/custom/ScopedRouteTypeTest.scala
package kpn.api.custom

import kpn.api.common.RouteScope
import kpn.api.common.RouteType
import kpn.core.util.UnitTest

class ScopedRouteTypeTest extends UnitTest {

  test("all") {
    ScopedRouteType.all should equal(
      Seq(
        ScopedRouteType(RouteType.hiking, RouteScope.local, "lwn"),
        ScopedRouteType(RouteType.hiking, RouteScope.regional, "rwn"),
        ScopedRouteType(RouteType.hiking, RouteScope.national, "nwn"),
        ScopedRouteType(RouteType.hiking, RouteScope.international, "iwn"),
        ScopedRouteType(RouteType.cycling, RouteScope.local, "lcn"),
        ScopedRouteType(RouteType.cycling, RouteScope.regional, "rcn"),
        ScopedRouteType(RouteType.cycling, RouteScope.national, "ncn"),
        ScopedRouteType(RouteType.cycling, RouteScope.international, "icn"),
        ScopedRouteType(RouteType.horseRiding, RouteScope.local, "lhn"),
        ScopedRouteType(RouteType.horseRiding, RouteScope.regional, "rhn"),
        ScopedRouteType(RouteType.horseRiding, RouteScope.national, "nhn"),
        ScopedRouteType(RouteType.horseRiding, RouteScope.international, "ihn"),
        ScopedRouteType(RouteType.canoe, RouteScope.local, "lpn"),
        ScopedRouteType(RouteType.canoe, RouteScope.regional, "rpn"),
        ScopedRouteType(RouteType.canoe, RouteScope.national, "npn"),
        ScopedRouteType(RouteType.canoe, RouteScope.international, "ipn"),
        ScopedRouteType(RouteType.motorboat, RouteScope.local, "lmn"),
        ScopedRouteType(RouteType.motorboat, RouteScope.regional, "rmn"),
        ScopedRouteType(RouteType.motorboat, RouteScope.national, "nmn"),
        ScopedRouteType(RouteType.motorboat, RouteScope.international, "imn"),
        ScopedRouteType(RouteType.inlineSkating, RouteScope.local, "lin"),
        ScopedRouteType(RouteType.inlineSkating, RouteScope.regional, "rin"),
        ScopedRouteType(RouteType.inlineSkating, RouteScope.national, "nin"),
        ScopedRouteType(RouteType.inlineSkating, RouteScope.international, "iin")
      )
    )
  }
}

File: backend/api/src/test/java/kpn/api/custom/ScopedRouteTypeTest.java
package kpn.api.custom;

import kpn.api.common.RouteScope;
import kpn.api.common.RouteType;
import org.junit.jupiter.api.Test;

import com.google.common.collect.ImmutableList;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ScopedRouteTypeTest {

  @Test
  public void all() {
    assertEquals(
      ImmutableList.of(
        new ScopedRouteType(RouteType.HIKING, RouteScope.LOCAL, "lwn"),
        new ScopedRouteType(RouteType.HIKING, RouteScope.REGIONAL, "rwn"),
        new ScopedRouteType(RouteType.HIKING, RouteScope.NATIONAL, "nwn"),
        new ScopedRouteType(RouteType.HIKING, RouteScope.INTERNATIONAL, "iwn"),
        new ScopedRouteType(RouteType.CYCLING, RouteScope.LOCAL, "lcn"),
        new ScopedRouteType(RouteType.CYCLING, RouteScope.REGIONAL, "rcn"),
        new ScopedRouteType(RouteType.CYCLING, RouteScope.NATIONAL, "ncn"),
        new ScopedRouteType(RouteType.CYCLING, RouteScope.INTERNATIONAL, "icn"),
        new ScopedRouteType(RouteType.HORSE_RIDING, RouteScope.LOCAL, "lhn"),
        new ScopedRouteType(RouteType.HORSE_RIDING, RouteScope.REGIONAL, "rhn"),
        new ScopedRouteType(RouteType.HORSE_RIDING, RouteScope.NATIONAL, "nhn"),
        new ScopedRouteType(RouteType.HORSE_RIDING, RouteScope.INTERNATIONAL, "ihn"),
        new ScopedRouteType(RouteType.CANOE, RouteScope.LOCAL, "lpn"),
        new ScopedRouteType(RouteType.CANOE, RouteScope.REGIONAL, "rpn"),
        new ScopedRouteType(RouteType.CANOE, RouteScope.NATIONAL, "npn"),
        new ScopedRouteType(RouteType.CANOE, RouteScope.INTERNATIONAL, "ipn"),
        new ScopedRouteType(RouteType.MOTORBOAT, RouteScope.LOCAL, "lmn"),
        new ScopedRouteType(RouteType.MOTORBOAT, RouteScope.REGIONAL, "rmn"),
        new ScopedRouteType(RouteType.MOTORBOAT, RouteScope.NATIONAL, "nmn"),
        new ScopedRouteType(RouteType.MOTORBOAT, RouteScope.INTERNATIONAL, "imn"),
        new ScopedRouteType(RouteType.INLINE_SKATING, RouteScope.LOCAL, "lin"),
        new ScopedRouteType(RouteType.INLINE_SKATING, RouteScope.REGIONAL, "rin"),
        new ScopedRouteType(RouteType.INLINE_SKATING, RouteScope.NATIONAL, "nin"),
        new ScopedRouteType(RouteType.INLINE_SKATING, RouteScope.INTERNATIONAL, "iin"),
        new ScopedRouteType(RouteType.MTB, RouteScope.LOCAL, "l?n"),
        new ScopedRouteType(RouteType.MTB, RouteScope.REGIONAL, "r?n"),
        new ScopedRouteType(RouteType.MTB, RouteScope.NATIONAL, "n?n"),
        new ScopedRouteType(RouteType.MTB, RouteScope.INTERNATIONAL, "i?n")
      ),
      ScopedRouteType.all()
    );
  }
}
